### PR TITLE
🚀 add patch for node v12.18.4

### PR DESCRIPTION
**Pull request by n1ru4l**

### CODE REVIEW

File: patches/patches.json [hunk 1]
```diff
@@ -8,6 +8,9 @@
   "v13.12.0": [
     "node.v13.12.0.cpp.patch"
   ],
+  "v12.18.4": [
+    "node.v12.18.4.cpp.patch"
```
Comment (robertsLando) [inline patches/patches.json:12]: @n1ru4l If the patch is the same of 12.18.3, what about just point the 12.18.3 here?

Comment (n1ru4l) [reply patches/patches.json:12]: I just replaced `12.18.3` completely with `12.18.4` (this is a different branch than the one we are using :))

Comment (robertsLando) [reply patches/patches.json:12]: Ok ignore my comment so :)

File: patches/node.v12.18.4.cpp.patch [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+ const { safeGetenv } = internalBinding('credentials');
+ const {
+--- node/lib/internal/modules/package_json_reader.js
++++ node/lib/internal/modules/package_json_reader.js
+@@ -1,7 +1,7 @@
+ 'use strict';
+ 
+ const { SafeMap } = primordials;
+-const { internalModuleReadJSON } = internalBinding('fs');
++const internalModuleReadJSON = function (f) { return require('fs').internalModuleReadJSON(f); };
```
Comment (dio) [inline patches/node.v12.18.4.cpp.patch:254]: Not sure but it seems like this gives: 

```
pkg/prelude/bootstrap.js:1244
      throw error;
      ^

TypeError: internalModuleReadJSON is not a function or its return value is not iterable
    at Object.read (internal/modules/package_json_reader.js:17:34)
    at readPackage (internal/modules/cjs/loader.js:249:36)
    at readPackageScope (internal/modules/cjs/loader.js:287:19)
    at trySelf (internal/modules/cjs/loader.js:396:41)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:942:22)
    at Function.Module._resolveFilename (pkg/prelude/bootstrap.js:1338:44)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at Module.require (pkg/prelude/bootstrap.js:1225:31)
    at require (internal/modules/cjs/helpers.js:72:18)pkg/prelude/bootstrap.js:1244
      throw error;
      ^

TypeError: internalModuleReadJSON is not a function or its return value is not iterable
    at Object.read (internal/modules/package_json_reader.js:17:34)
    at readPackage (internal/modules/cjs/loader.js:249:36)
    at readPackageScope (internal/modules/cjs/loader.js:287:19)
    at trySelf (internal/modules/cjs/loader.js:396:41)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:942:22)
    at Function.Module._resolveFilename (pkg/prelude/bootstrap.js:1338:44)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at Module.require (pkg/prelude/bootstrap.js:1225:31)
    at require (internal/modules/cjs/helpers.js:72:18)
```

Comment (rahbari) [reply patches/node.v12.18.4.cpp.patch:254]:
```
TypeError: internalModuleReadJSON is not a function or its return value is not iterable
    at Object.read (internal/modules/package_json_reader.js:21:34)
    at readPackage (internal/modules/cjs/loader.js:264:36)
    at readPackageScope (internal/modules/cjs/loader.js:297:19)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1087:17)
    at Module.load (internal/modules/cjs/loader.js:941:32)
    at Function.Module._load (internal/modules/cjs/loader.js:782:14)
    at Function.Module.runMain (pkg/prelude/bootstrap.js:1375:12)
    at internal/main/run_main_module.js:17:47
```

Comment (n1ru4l) [reply patches/node.v12.18.4.cpp.patch:254]: can you come up with a fix?

**Comment by rand256** [comment]
I've compiled it for armv7 and the resulting app can't start complaining that it can't parse the main package.json that is perfectly valid:

```
# ./myapp
internal/modules/cjs/loader.js:274
    throw e;
    ^

SyntaxError: Error parsing /snapshot/myapp/package.json: Unexpected end of JSON input
    at parse (<anonymous>)
    at readPackage (internal/modules/cjs/loader.js:262:20)
    at readPackageScope (internal/modules/cjs/loader.js:287:19)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1148:17)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Function.Module.runMain (pkg/prelude/bootstrap.js:1375:12)
    at internal/main/run_main_module.js:17:47 {
  path: '/snapshot/myapp/package.json'
```
It looks like the only opening brace character is read from the file somehow, so obviously the input's end is unexpected.

**Comment by alexk111** [comment]
```
TypeError: internalModuleReadJSON is not a function or its return value is not iterable
````

In newer versions of Node.js the `internalModuleReadJSON` function returns an array instead of a string. To fix the mentioned issue pkg's `bootstrap.js` should also be updated. Here is the fixed one: https://github.com/alexk111/pkg/commit/0a9eefbc88e3a2d89ef6d0a863122b1fe94d96b6

**Comment by n1ru4l** [comment]
If anyone wanna pick this up feel free. I am going to fully migrate over to caxa once it supports ARM https://github.com/leafac/caxa/issues/4

**Comment by xanonid** [comment]
I am working on a node v12.21.0 patch based on this PR.

**Comment by leerob** [comment]
Thank you @xanonid! 🙏

**Comment by david-mohr** [comment]
I have a fix for the `internalModuleReadJSON` function in #118 

```
--- node/lib/internal/modules/package_json_reader.js
+++ node/lib/internal/modules/package_json_reader.js
@@ -1,7 +1,7 @@
 'use strict';

 const { SafeMap } = primordials;
-const { internalModuleReadJSON } = internalBinding('fs');
+const internalModuleReadJSON = function (f) { let json = require('fs').internalModuleReadJSON(f); return [json, !!json]; };
 const { pathToFileURL } = require('url');
 const { toNamespacedPath } = require('path');
```

**Pull request closed without merging**